### ISSUE TITLE
Пропадают объекты медиа

### ISSUE THREAD

**Issue by venticello** [livestreet/livestreet#717]
Ошибка повторилась, просто открыл чужой топик под админом, подправил заголовок, больше ничего не менял, нажал сохранить и ссылки на медиа топика потерялись. Превью у топика сохранилось.

**Comment by mzhelskiy** [comment]
Т.е. у тебя при сохранении топика target_id стал null, а target_tmp принял значение 7c6b86ed06?

**Comment by mzhelskiy** [comment]
Сейчас устроено так: для нового топика линки на медиа идут с target_tmp (он прописывается только в момент загрузки фото), далее после создания прописывается target_id, а target_tmp становится null. Каким образом у тебя опять прописывается (уже при редактировании) target_tmp не понятно, т.к. в коде только одно место, где это значение устанавливается - это момент загрузки конкретного медиа файла.

**Comment by venticello** [comment]
Я не отслеживал какое значение до этого было в  target_tmp. И странно как оно обратно может прописаться, такую ситуацию замечал только когда чужой топик редактируется. Причем раньше думал, что потому что открывал медиа, а в этот раз просто заголовок поменял.

**Comment by venticello** [comment]
Повторилась ситуация, редактирование, просто добавил cut и медиа также отвалились, буду следить далее.

**Comment by venticello** [comment]
До сохранения топика отображаются медиа, но в превью уже не видно, в базе медиа target_tmp, нет 	target_id т.е. медиа изначально как то попали в топик, но не привязались к топику. Как такое может произойти?

**Comment by mzhelskiy** [comment]
Нового топика или при редактировании уже созданного?
Не показывается только в превью? После сохранения они привязываются к посту?

**Comment by venticello** [comment]
Кажется понял, вариант такой, создаю топик, загружаю туда фотки, но не сохраняю.
Потом беру создаю новый топик, или редактирую еще один, вставляю туда фото из списка всех моих медиа и все, ссылка не проставляется на топик.

**Comment by mzhelskiy** [comment]
Попробовал вот так https://github.com/livestreet/livestreet/commit/670b70f4bb6e7890e9f1a0dc9f3106f1d1dccab6

**Comment by venticello** [comment]
Это получается пользователю не будут показываться его Медиа созданные и не привязанные к топикам? Они получается зависнут в темпах или я не правильно понял? 
Проблема по сути в том, что админ не видит не привязанные объекты и они отваливаются при редакитровании топика.

**Comment by mzhelskiy** [comment]
Да, получается, что это временные данные. Их нужно будет удалять по крону.
Про "отваливаются при редакитровании топика" не понял.

**Comment by venticello** [comment]
Это значит, до редактирования админом, эти медиа спокойно отображаются в топике, а если админ будет редактировать, то он их не находит и не отображает.
По крону, а если пользователь их вставит, получится, после чистки кроном, они в итоговом топике пропадут.

**Comment by mzhelskiy** [comment]
Подожди, если мы редактируем топик, то медиа у него уже привязаны и такой проблемы не будет. Юзеру будет проблемно вставить в топик временные медиа не своего топика, а ели своего, то после создания они перепривяжутся и уже не будут временными.

**Comment by venticello** [comment]
Если он конечно не увидит свои предыдущие временные загрузки, тогда да проблемы не будет. Думаю вряд ли кто то будет копировать из временного топика вставленные изображения, в тот который будет публиковать.

**Comment by mzhelskiy** [comment]
Ну вот, прошлым коммитом как раз исключил из показа временные загрузки.
